app/presenters/date_question_presenter.rb

```diff
@@ -12,22 +12,18 @@ def response_label(value)
     end
   end
 
-  def selected_day
-    return unless response_for_current_question.is_a? Hash
-
-    response_for_current_question[:day]
-  end
-
-  def selected_month
-    return unless response_for_current_question.is_a? Hash
-
-    response_for_current_question[:month]
-  end
-
-  def selected_year
-    return unless response_for_current_question.is_a? Hash
-
-    response_for_current_question[:year]
+  def parsed_response
+    return {} if current_response.blank?
+    return current_response if current_response.is_a?(Hash)
+
+    date = @node.parse_input(current_response)
+    {
+      day: date.day,
+      month: date.month,
+      year: date.year,
+    }
+  rescue SmartAnswer::InvalidResponse
+    {}
```

Comment:
Does this suppress an error being shown to the user, if current response is invalid?

Reply:
Nope, this is the value that is shown to the user in the UI to the user with their error. Having it returns an empty hash means that the view only needs to be concerned that this presenter will return one type whether an error input or not.